hypertrace-trace-enricher/enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanHeaderUtils.java

```diff
@@ -0,0 +1,122 @@
+package org.hypertrace.traceenricher.enrichedspan.constants.utils;
+
+import static com.google.common.net.HttpHeaders.COOKIE;
+import static com.google.common.net.HttpHeaders.SET_COOKIE;
+import static org.hypertrace.core.span.constants.v1.Grpc.GRPC_REQUEST_METADATA;
+import static org.hypertrace.core.span.constants.v1.Grpc.GRPC_RESPONSE_METADATA;
+import static org.hypertrace.core.span.constants.v1.Http.HTTP_REQUEST_HEADER;
+import static org.hypertrace.core.span.constants.v1.Http.HTTP_RESPONSE_HEADER;
+import static org.hypertrace.core.span.normalizer.constants.RpcSpanTag.RPC_REQUEST_METADATA;
+import static org.hypertrace.core.span.normalizer.constants.RpcSpanTag.RPC_RESPONSE_METADATA;
+import static org.hypertrace.semantic.convention.utils.rpc.RpcSemanticConventionUtils.isRpcSystemGrpc;
+import static org.hypertrace.traceenricher.enrichedspan.constants.utils.EnrichedSpanUtils.isEmptyAttributesMap;
+import static org.hypertrace.traceenricher.enrichedspan.constants.utils.EnrichedSpanUtils.isValueNotNull;
+
+import com.google.common.base.Predicate;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.span.constants.RawSpanConstants;
+
+public class EnrichedSpanHeaderUtils {
```

Comment:
Similarly, can we instead add the util functions inside `HttpSemanticConventionUtils` and an umbrella function under `EnrichedSpanUtils`?

Reply:
We are using it for getting headers from GRPC requests, so we need to duplicate it for RpcSemanticConventionUtils?

Reply:
Initially, I'd asked Shivansh to introduce 2 new utils @ https://github.com/hypertrace/hypertrace-ingester/pull/338#discussion_r962764685 for maintainable size of classes since they both can be easily separated.

Reply:
> We are using it for getting headers from GRPC requests, so we need to duplicate it for RpcSemanticConventionUtils?

Yeah. We can add one for HTTP in HttpSemanticConventionUtils, one for gRPC in RpcSemanticConventionUtils and, then a combined one in EnrichedSpanUtils